### PR TITLE
sfml/csfml 2.3

### PR DESCRIPTION
**Pull request by dunn**
The last update was a bit tricky (#35279) so let me know if there's anything I overlooked here.

### CODE REVIEW

**Comment by DomT4** [comment]
I knew I'd forgotten something. They've made major changes to the deps, see https://github.com/SFML/SFML/pull/852#issuecomment-99855970.

**Comment by dunn** [comment]
From that thread:
> Also on OS X we don't use the default OpenAL shipped by Apple.
Does that mean `openal-soft` for us?

**Comment by DomT4** [comment]
Looking at what upstream do for other platforms, yup.

**Comment by mikemcquaid** [comment]
`csfml` failing: http://bot.brew.sh/job/Homebrew%20Pull%20Requests/26400/version=yosemite/testReport/junit/brew-test-bot/yosemite/install_csfml/

**Comment by dunn** [comment]
They're recommending we wait for 2.3 of CSFML

**Comment by mantognini** [comment]
@dunn Thanks for updating those! :-)

File: Library/Formula/sfml.rb [hunk 1]
```diff
@@ -21,26 +12,43 @@ class Sfml < Formula
...
-    # https://github.com/Homebrew/homebrew/pull/35279
-    rm_rf buildpath/"extlibs"
+    # Always remove the "extlibs" to avoid install_name_tool failure
+    # (https://github.com/Homebrew/homebrew/pull/35279) but leave the
+    # headers that were moved there in https://github.com/SFML/SFML/pull/795
+    Dir.glob("extlibs/*") do |dir|
+      if dir != "extlibs/headers"
+        rm_rf dir
+      end
+    end
```
Comment (jacknagel) [inline Library/Formula/sfml.rb]: `rm_rf Dir["extlibs/*"] - ["extlibs/headers"]`?

Comment (dunn) [inline Library/Formula/sfml.rb]: thanks!

**Comment by dunn** [comment]
Is an update warning at the top of each formula appropriate?

**Comment by mikemcquaid** [comment]
> Is an update warning at the top of each formula appropriate?

:+1:

**Comment by dunn** [comment]
Linking CSFML fails because it has some of the same headers SFML has:
```
Error: The `brew link` step did not complete successfully
The formula built, but is not symlinked into /usr/local
Could not symlink include/SFML/Audio/AlResource.hpp
Target /usr/local/include/SFML/Audio/AlResource.hpp
is a symlink belonging to sfml. You can unlink it:
  brew unlink sfml

To force the link and overwrite all conflicting files:
  brew link --overwrite csfml

To list all files that would be deleted:
  brew link --overwrite --dry-run csfml

Possible conflicting files are:
/usr/local/include/SFML/Audio/AlResource.hpp -> /usr/local/Cellar/sfml/2.3/include/SFML/Audio/AlResource.hpp
```

Would it be safe to just remove CSFML's `include/SFML` directory, @mantognini?

**Comment by mantognini** [comment]
That's a strange error you got there... there is no `AlResource.hpp` in CSFML as far as I know.

**Comment by DomT4** [comment]
> Linking CSFML fails because it has some of the same headers SFML has:

You seem to be using the same tarball for the two formulae? Were the two tarballs merged?

**Comment by dunn** [comment]
ahahahaha, my bad.

**Comment by DomT4** [comment]
Is this ready to ship? Not waiting on anything else now?

**Comment by dunn** [comment]
@mantognini, should we wait until the CSFML 2.3 source is available from http://www.sfml-dev.org/download/csfml/ ?

**Comment by mantognini** [comment]
For CSFML you can safely use GitHub zip archive, yes.

For SFML, we remove some files from it so it's safer to use the one from sfml-dev.org (currently we only remove .gitattributes but it used to be more than that -- don't know about the future though ;-))

**Comment by DomT4** [comment]
:ship:'d in https://github.com/Homebrew/homebrew/commit/e2de0389ed51aef47f8f67c9e67c4db79dde88e2. Thanks @dunn @mantognini!

I removed the `desc` lines manually before merge as Mike requested we don't add any to existing formulae for now. Side note: Separate commits per formula (Inside the same PR) in this situation is preferable.

**Comment by dunn** [comment]
Good to know, thanks.

**Pull request closed without merging**